packages/Discretization/src/DTK_NearestNeighborOperator_decl.hpp

```diff
@@ -0,0 +1,43 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_NEAREST_NEIGHBOR_OPERATOR_DECL_HPP
+#define DTK_NEAREST_NEIGHBOR_OPERATOR_DECL_HPP
+
+#include <DTK_ConfigDefs.hpp>
+
+#include <Kokkos_Core.hpp>
+#include <Teuchos_Comm.hpp>
+
+namespace DataTransferKit
+{
+
+template <typename DeviceType>
+class NearestNeighborOperator
+{
+    using ExecutionSpace = typename DeviceType::execution_space;
+
+  public:
+    NearestNeighborOperator(
+        Teuchos::RCP<const Teuchos::Comm<int>> const &comm,
+        Kokkos::View<Coordinate **, DeviceType> const &source_points,
```

Comment:
Is this what we want or should we use `View<Coordinate const**, DeviceType> const &`?

Reply:
Then you would not be able to pass a `View<Coordinate **, DeviceType>` as an argument.